pages/api/auth/[...nextauth].ts

```diff
@@ -174,6 +178,101 @@ const createUserUpdateObj = (user: User, account: Account | null, profile: Profi
 	return updateObj;
 }
 
+const sendSignupEmail = (userEmail: string, userName: string) => {
+	const htmlBody = `
+<!DOCTYPE html>
+<html lang="en">
+<head>
+	<meta charset="UTF-8">
+	<meta name="viewport" content="width=device-width, initial-scale=1.0">
+	<title>Welcome to Vibinex</title>
+	<style>
+		/* Add your custom CSS styles here */
+		body {
+			font-family: Arial, sans-serif;
+			background-color: #f5f5f5;
+			color: #333;
+			margin: 0;
+			padding: 0;
+		}
+		.container {
+			max-width: 600px;
+			margin: 20px auto;
+			padding: 20px;
+			background-color: #fff;
+			border-radius: 8px;
+			box-shadow: 0 0 10px rgba(0, 0, 0, 0.1);
+		}
+		.logo {
+			text-align: center;
+			margin-bottom: 20px;
+		}
+		.logo img {
+			max-width: 150px;
+			height: auto;
+		}
+		h1 {
+			color: #2196F3;
+			text-align: center;
+		}
+		p {
+			margin-bottom: 20px;
+			line-height: 1.5;
+		}
+		.cta-button {
+			display: block;
+			width: 200px;
+			margin: 20px auto;
+			padding: 10px 20px;
+			text-align: center;
+			background-color: #2196F3;
+			color: #fff;
+			text-decoration: none;
+			border-radius: 4px;
+			transition: background-color 0.3s ease;
+		}
+		.cta-button:hover {
+			background-color: #0c7cd5;
+		}
+	</style>
+</head>
+<body>
+	<div class="container">
+		<div class="logo">
+			<img src="https://vibinex.com/favicon.ico" alt="Vibinex Logo">
+		</div>
+		<h1>Welcome to Vibinex, ${userName}!</h1>
+		<p>With Vibinex, you can streamline your pull request reviews, collaborate with your team more efficiently, and accelerate your software development process.</p>
+		<p>Start exploring now by logging into your account:</p>
+		<a href="https://vibinex.com/login" class="cta-button">Log In to Vibinex</a>
+		<p>If you have any questions or need assistance, feel free to reach out to our support team at <a href="mailto:contact@vibinex.com">contact@vibinex.com</a>.</p>
+		<p>Happy coding!</p>
+		<p>The Vibinex Team</p>
+	</div>
+</body>
+</html>
+	`;
+	console.debug(`[sendSignupEmail] SENDGRID_API_KEY = ${process.env.SENDGRID_API_KEY}`);
+	console.debug(`[sendSignupEmail] user email = ${userEmail}`);
+	const msg = {
+		to: userEmail, // recipient's email address
+		from: "contact@vibinex.com", // sender's email address
+		subject: "Welcome to Vibinex", // email subject
+		html: htmlBody,
+	};
+	// Send email
+	if (!process.env.SENDGRID_API_KEY) {
+		console.error("[sendSignupEmail] SENDGRID_API_KEY env var not set, unable to send signup email");
```

Comment:
Can we add some identification here so that we can identify the sign-up for which the email failed? The user id or email would be good enough.

Reply:
(Low LOGAF, but remember to also add eventing afterwards.)